### PR TITLE
Tweak main_navigation block to align with designs

### PR DESCRIPTION
**Pull request by AshGDS**
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

## What / Why
- Makes the margin between the `Service name` (black link) and the menu items (blue links) the same.
- Adds more spacing to the mobile menu, and adds left padding to the list items so that they are staggered away from the menu button, as per the designs.
- Adds a `border_bottom` to the block. This required turning it into a proper block, so that `full_width?` could be used, otherwise the border would only go up to the width limits of the `govuk-width-container` class. 
- Renames the `children:` variable for child links to `items:`. I tried to get it work with `children:` as thats more readable, but for some reason it didn't work. Might be a backend thing going over my head.
- Trello: https://trello.com/c/BmgLrgZD / https://trello.com/c/hzfHN1fI

### Before
image
image

### After

image

image

### CODE REVIEW

File: app/views/landing_page/blocks/_main_navigation.html.erb
Comment (AshGDS) [inline app/views/landing_page/blocks/_main_navigation.html.erb:45]: For ease of review as the diff looks a bit crazy, all I've done in this file is:
- added `<div class="app-b-main-nav--border">` around the block
- added `govuk-width-container` to the first child div
- switched `block.data["title"]` to `block.title`, `block.data["children"] to `block.items`, etc.

File: app/views/landing_page/blocks/_main_navigation.html.erb
Comment (andysellick) [inline app/views/landing_page/blocks/_main_navigation.html.erb:5]: Seems like the BEM's got a bit confusing here... `app-b-main-nav--border` is a modifier class, but it's being used as the parent class. I think the first class that wraps the whole block should be `app-b-main-nav` and then the inner div (with the govuk width container) should be an element class, like `app-b-main-nav__wrapper` or something. Will need to juggle the actual styles, obviously.

Comment (andysellick) [inline app/views/landing_page/blocks/_main_navigation.html.erb:5]: I'd also put the `data-module` on the parent, if that's not going to cause a problem.

File: app/assets/stylesheets/views/_landing_page/main-navigation.scss
Comment (andysellick) [inline app/assets/stylesheets/views/_landing_page/main-navigation.scss:1]: The focus state for desktop links seems to be offset up by about a pixel? Not sure if that's intentional?

Screenshot 2024-10-23 at 09 40 55

Comment (AshGDS) [reply app/assets/stylesheets/views/_landing_page/main-navigation.scss:1]: Thanks, fixed.

File: app/views/landing_page/blocks/_main_navigation.html.erb
Comment (AshGDS) [reply app/views/landing_page/blocks/_main_navigation.html.erb]: Thanks, done 👍

File: spec/models/block/main_navigation_spec.rb
Comment (leenagupte) [inline spec/models/block/main_navigation_spec.rb:28]: I love that you've added model tests ⭐

File: app/models/block/main_navigation.rb
Comment (leenagupte) [inline app/models/block/main_navigation.rb:8]: To be honest, I'm not sure if this model is really needed as the you can just use `block.data["title"]` etc in the views. However being able to call `block.title` is tidier.

**Review by leenagupte — COMMENTED** [review]
This looks good from a backend perspective. 👍 

As there's a lot of frontendy changes, I'll let Andy approve.

**Comment by AshGDS** [comment]
Thanks both 👍 

@andysellick - the frontend should be ready for re-review

File: app/assets/stylesheets/views/_landing_page/main-navigation.scss
Comment (andysellick) [reply app/assets/stylesheets/views/_landing_page/main-navigation.scss:1]: The issue's fixed although I notice it's changed the appearance of the focus state so that it no longer reaches to the top of the element. I think this is fine and shouldn't block this PR, but I've realised that I can't see any focus states in the designs so we should get this checked separately by design.

**Review by andysellick — APPROVED** [review]
Frontend looks good to me 👍

**Pull request merged**
